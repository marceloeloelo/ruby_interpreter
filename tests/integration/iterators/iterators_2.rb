var = 1

encontre = false
while ( var && var && !encontre )
  encontre = true
  puts ("bien")
end

encontre = false
while ( (var || var) && !encontre)
  encontre = true
  puts ("bien")
end

while ( var && nil )
  puts ("mal")
end

encontre = false
while ( (var || nil) && !encontre )
  encontre = true
  puts ("bien")
end

encontre = false
while ( !(var && nil || nil && 1) && !encontre )
  encontre = true
  puts ("bien")
end

encontre = false
while ( (var >= 1 || var <= 1) && !encontre )
  encontre = true
  puts ("bien")
end

encontre = false
while ( (var > 5 && var >= 10 && var == nil || var) &&  !encontre )
  encontre = true
  puts ("bien")
end

encontre = false
cont = 0
while (!encontre)
  if ( (var > 5 && var >= 10 && var == nil || var) && cont == 3 )
	encontre = true
  end
  cont += 1
  puts (cont)
end

