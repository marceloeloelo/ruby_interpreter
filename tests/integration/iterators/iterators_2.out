bien
bien
bien
bien
bien
bien
1
2
3
4
